### PR TITLE
[Docs] Reorganize online serving docs.

### PR DESCRIPTION
**Pull request by noooop**
## Purpose

In the early days of vLLM (2023), there was only the OpenAI, so vLLM online serving was referred to as the OpenAI-Compatible Server. 

As we have entered 2026, generative AI has grown significantly, and more and more features have been added to vLLM online serving. 

Therefore, it is necessary for us to reorganize the online serving documentation.

I feel that many online serving interfaces and features have not been documented.  

Let's supplement them in the upcoming PRs.

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

File: docs/getting_started/quickstart.md [hunk 1]
```diff
@@ -167,7 +167,7 @@ for output in outputs:
             print(f"Prompt: {prompt!r}, Generated text: {generated_text!r}")
         ```
 
-## OpenAI-Compatible Server
+## Online Serving
```
Comment (DarkLight1337) [inline docs/getting_started/quickstart.md:185]: Can you add redirects from `serving/openai_compatible_server` to the new location. in `mkdocs.yaml`?

File: docs/serving/online_serving/generative_scoring_api.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# Generative Scoring API
```
Comment (DarkLight1337) [inline docs/serving/online_serving/generative_scoring_api.md]: Remove "API" from the file name to be consistent with other files

File: docs/serving/online_serving/others.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Others APIs
```
Comment (DarkLight1337) [inline docs/serving/online_serving/others.md]: Let's just put tokenizer API as its own page, don't use "others"

File: docs/serving/online_serving/render.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# vLLM render APIs
```
Comment (DarkLight1337) [inline docs/serving/online_serving/render.md]:
```suggestion
# Renderer APIs
```

File: docs/serving/online_serving/renderer.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+- GPU-less deployment of frontend: Allow preprocessing (tokenization, MM input processing) and postprocessing (detokenization, tool call parsing, reasoning parsing) to run without GPU.
+- Disaggregated tokenization: Support use cases such as llm-d, Dynamo, and custom frontends that need to leverage vLLM's preprocessing logic without running the full inference engine.
+- Tokens-in / tokens-out engine: Make the engine a pure token-in / token-out service, decoupled from request preprocessing.
+
+## API Reference
+
+- Completions render API (`/v1/completions/render`)
+    - Render completion requests
+- Chat Completions render API (`/v1/chat/completions/render`)
+    - Render Chat Completions
```
Comment (noooop) [inline docs/serving/online_serving/renderer.md]: cc @hyeongyun0916

Could you please add some content for the Renderer APIs, after this PR is merged.

File: docs/serving/online_serving/README.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+    - Only applicable to [Automatic Speech Recognition (ASR) models](../../models/supported_models.md#transcription).
+- [Translation API](./speech_to_text.md#translations-api) (`/v1/audio/translations`)
+    - Only applicable to [Automatic Speech Recognition (ASR) models](../../models/supported_models.md#transcription).
+- [Realtime API](./speech_to_text.md#realtime-api) (`/v1/realtime`)
+    - Only applicable to [Automatic Speech Recognition (ASR) models](../../models/supported_models.md#realtime-transcription).
+
+## Anthropic-compatible messages API
+
+- Anthropic messages API (`/v1/messages`)
+
```
Comment (noooop) [inline docs/serving/online_serving/README.md]: cc @chaunceyjiang @sfeng33 

Could you please add some content for Anthropic, after this PR is merged.

File: docs/serving/online_serving/renderer.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+The vLLM render API is designed to disaggregate the render phase(preprocessing) and enable a token-in / token-out API server.
+
+- GPU-less deployment of frontend: Allow preprocessing (tokenization, MM input processing) and postprocessing (detokenization, tool call parsing, reasoning parsing) to run without GPU.
+- Disaggregated tokenization: Support use cases such as llm-d, Dynamo, and custom frontends that need to leverage vLLM's preprocessing logic without running the full inference engine.
+- Tokens-in / tokens-out engine: Make the engine a pure token-in / token-out service, decoupled from request preprocessing.
+
+## API Reference
+
+- Completions render API (`/v1/completions/render`)
```
Comment (DarkLight1337) [inline docs/serving/online_serving/renderer.md]: Please make sure the capitalization is consistent

File: docs/serving/online_serving/README.md [hunk 8]
```diff
@@ -0,0 +1,178 @@
+# Online Serving
+
+vLLM provides an HTTP server that is compatible with many interfaces!
+
+## OpenAI-Compatible Server
```
Comment (DarkLight1337) [inline docs/serving/online_serving/README.md:5]: Also check the capitalization on this page

File: mkdocs.yaml [hunk 9]
```diff
@@ -110,6 +110,7 @@ plugins:
       redirect_maps:
         features/spec_decode/README.md: features/speculative_decoding/README.md
         features/spec_decode/speculators.md: features/speculative_decoding/speculators.md
+        serving/openai_compatible_server: serving/online_serving/README.md
```
Comment (hmellor) [inline mkdocs.yaml]:
```suggestion
        serving/openai_compatible_server.md: serving/online_serving/README.md
```

File: docs/serving/online_serving/README.md [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+
+Ray Serve LLM enables scalable, production-grade serving of the vLLM engine. It integrates tightly with vLLM and extends it with features such as auto-scaling, load balancing, and back-pressure.
+
+Key capabilities:
+
+- Exposes an OpenAI-compatible HTTP API as well as a Pythonic API.
+- Scales from a single GPU to a multi-node cluster without code changes.
+- Provides observability and autoscaling policies through Ray dashboards and metrics.
+
+The following example shows how to deploy a large model like DeepSeek R1 with Ray Serve LLM: [examples/ray_serving/ray_serve_deepseek.py](../../../examples/ray_serving/ray_serve_deepseek.py).
```
Comment (hmellor) [inline docs/serving/online_serving/README.md:180]: This path doesn't exist, did you mean `examples/online_serving/ray_serve_deepseek.py`?

Comment (noooop) [reply docs/serving/online_serving/README.md:180]: We have finally completed the refactoring of the examples folder. 

#41082  #41084

Below is the new directory structure.

```
├─applications
│  ├─chatbot
│  └─rag
├─basic
│  ├─offline_inference
│  └─online_serving
├─deployment
│  └─chart-helm
│      ├─templates
│      └─tests
├─disaggregated
│  ├─disaggregated_encoder
│  ├─disaggregated_serving
│  ├─ec_both_encoder
│  ├─example_connector
│  ├─flexkv_connector
│  ├─kv_load_failure_recovery_offline
│  ├─lmcache
│  │  └─disagg_prefill_lmcache_v1
│  │      └─configs
│  ├─mooncake_connector
│  └─p2p_nccl_xpyd
├─features
│  ├─automatic_prefix_caching
│  ├─batch_invariance
│  ├─context_extension
│  ├─data_parallel
│  ├─kv_events
│  ├─logits_processor
│  ├─lora
│  ├─openai_batch
│  ├─pause_resume
│  ├─profiling
│  ├─prompt_embed
│  ├─reset_kv
│  ├─sharded_state
│  ├─speculative_decoding
│  ├─structured_outputs
│  └─torchrun
├─generate
│  └─multimodal
│      ├─qwen2_5_omni
│      └─qwen3_omni
├─observability
│  ├─dashboards
│  │  ├─grafana
│  │  └─perses
│  ├─metrics
│  ├─opentelemetry
│  └─prometheus_grafana
├─pooling
│  ├─classify
│  ├─embed
│  │  ├─openai_embedding_long_text
│  │  └─template
│  ├─plugin
│  ├─reward
│  ├─score
│  │  └─template
│  ├─token_classify
│  └─token_embed
├─ray_serving
│  └─elastic_ep
├─reasoning
├─rl
├─speech_to_text
│  ├─lid
│  ├─openai
│  └─realtime
└─tool_calling
```

File: docs/serving/online_serving/README.md [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+    - *Note: `user` parameter is ignored.*
+    - *Note:* Setting the `parallel_tool_calls` parameter to `false` ensures vLLM only returns zero or one tool call per request. Setting it to `true` (the default) allows returning more than one tool call per request. There is no guarantee more than one tool call will be returned if this is set to `true`, as that behavior is model dependent and not all models are designed to support parallel tool calls.
+- [Embeddings API](../../models/pooling_models/embed.md#openai-compatible-embeddings-api) (`/v1/embeddings`)
+    - Only applicable to [embedding models](../../models/pooling_models/embed.md).
+- [Transcriptions API](./speech_to_text.md#transcriptions-api) (`/v1/audio/transcriptions`)
+    - Only applicable to [Automatic Speech Recognition (ASR) models](../../models/supported_models.md#transcription).
+- [Translation API](./speech_to_text.md#translations-api) (`/v1/audio/translations`)
+    - Only applicable to [Automatic Speech Recognition (ASR) models](../../models/supported_models.md#transcription).
+
+## Anthropic-compatible messages API
```
Comment (DarkLight1337) [inline docs/serving/online_serving/README.md]:
```suggestion
## Anthropic APIs
```

File: docs/serving/online_serving/README.md [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+## Cohere-Compatible Server
+
+- [Cohere Embed API](../../models/pooling_models/embed.md#cohere-embed-api) (`/v2/embed`)
+    - Compatible with [Cohere's Embed API](https://docs.cohere.com/reference/embed)
+    - Works with any [embedding model](../../models/pooling_models/embed.md#supported-models), including multimodal models.
+- [Rerank API](../../models/pooling_models/scoring.md#rerank-api) (`/rerank`, `/v1/rerank`, `/v2/rerank`)
+    - Implements [Jina AI's v1 rerank API](https://jina.ai/reranker/)
+    - compatible with [Cohere's v1 & v2 rerank APIs](https://docs.cohere.com/v2/reference/rerank)
+
+## SageMaker-compatible
```
Comment (DarkLight1337) [inline docs/serving/online_serving/README.md]:
```suggestion
## SageMaker APIs
```

File: docs/serving/online_serving/README.md [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+- [Transcriptions API](./speech_to_text.md#transcriptions-api) (`/v1/audio/transcriptions`)
+    - Only applicable to [Automatic Speech Recognition (ASR) models](../../models/supported_models.md#transcription).
+- [Translation API](./speech_to_text.md#translations-api) (`/v1/audio/translations`)
+    - Only applicable to [Automatic Speech Recognition (ASR) models](../../models/supported_models.md#transcription).
+
+## Anthropic-compatible messages API
+
+- Anthropic messages API (`/v1/messages`)
+
+## Cohere-Compatible Server
```
Comment (DarkLight1337) [inline docs/serving/online_serving/README.md]:
```suggestion
## Cohere APIs
```

File: docs/serving/online_serving/README.md [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+    - Works with any [embedding model](../../models/pooling_models/embed.md#supported-models), including multimodal models.
+- [Rerank API](../../models/pooling_models/scoring.md#rerank-api) (`/rerank`, `/v1/rerank`, `/v2/rerank`)
+    - Implements [Jina AI's v1 rerank API](https://jina.ai/reranker/)
+    - compatible with [Cohere's v1 & v2 rerank APIs](https://docs.cohere.com/v2/reference/rerank)
+
+## SageMaker-compatible
+
+- `/invocations` - SageMaker-compatible endpoint (routes to the same inference functions as `/v1` endpoints)
+
+## Pooling models APIs
```
Comment (DarkLight1337) [inline docs/serving/online_serving/README.md]:
```suggestion
## Pooling APIs
```

File: docs/serving/online_serving/README.md [hunk 15]
```diff
@@ -0,0 +1,180 @@
...
+
+- Classification Usages
+    - [Classification API](../../models/pooling_models/classify.md#online-serving)(`/classify`)
+    - Only applicable to [classification models](../../models/pooling_models/classify.md).
+- Embedding Usages
+    - [Cohere Embed API](../../models/pooling_models/embed.md#cohere-embed-api) (`/v2/embed`)
+    - [Openai-compatible Embeddings API](../../models/pooling_models/embed.md#openai-compatible-embeddings-api) (`/v1/embeddings`)
+    - Only applicable to [embedding models](../../models/pooling_models/embed.md).
+- Scoring Usages
+    - [Score API](../../models/pooling_models/scoring.md#score-api)(/score)
```
Comment (DarkLight1337) [inline docs/serving/online_serving/README.md]:
```suggestion
    - [Score API](../../models/pooling_models/scoring.md#score-api) (`/score`)
```

File: docs/serving/online_serving/README.md [hunk 16]
```diff
@@ -0,0 +1,180 @@
...
+    - [Rerank API](../../models/pooling_models/scoring.md#rerank-api) (/rerank, /v1/rerank, /v2/rerank)
```
Comment (DarkLight1337) [inline docs/serving/online_serving/README.md]:
```suggestion
    - [Rerank API](../../models/pooling_models/scoring.md#rerank-api) (`/rerank`, `/v1/rerank`, `/v2/rerank`)
```

Comment (DarkLight1337) [reply docs/serving/online_serving/README.md]: Maybe we should also mention "Cohere" in the name (like for Embed API)

File: docs/serving/online_serving/README.md [hunk 17]
```diff
@@ -0,0 +1,180 @@
...
+- [Chat Completions Render API](renderer.md) (`/v1/chat/completions/render`)
+    - Render chat completions
+
+## Custom APIs
+
+- [Classification API](../../models/pooling_models/classify.md#classification-api) (`/classify`)
+    - Only applicable to [classification models](../../models/pooling_models/classify.md).
+- [Score API](../../models/pooling_models/scoring.md#score-api) (`/score`, `/v1/score`)
+    - Applicable to [score models](../../models/pooling_models/scoring.md) (cross-encoder, bi-encoder, late-interaction).
+- [pooling API](../../models/pooling_models/README.md#pooling-api) (`/pooling`)
```
Comment (DarkLight1337) [inline docs/serving/online_serving/README.md]:
```suggestion
- [Pooling API](../../models/pooling_models/README.md#pooling-api) (`/pooling`)
```

File: docs/serving/online_serving/README.md (hunk 14 above)
Comment (DarkLight1337) [reply docs/serving/online_serving/README.md]: Tbh I think we should avoid duplication with Cohere and Custom APIs

Comment (noooop) [reply docs/serving/online_serving/README.md]: Custom APIs add some pooling APIs to avoid having only the Generative Scoring API below. . . . 

According to this taxonomy, 
- the Score API (/score, /v1/score) is a Custom API, 
- while the Rerank API (/rerank, /v1/rerank, /v2/rerank) is a Cohere API.

Comment (DarkLight1337) [reply docs/serving/online_serving/README.md]: We can have classification, pooling score and generative score as custom apis

Comment (noooop) [reply docs/serving/online_serving/README.md]: That's just how it is.

File: docs/serving/online_serving/README.md [hunk 18]
```diff
@@ -0,0 +1,182 @@
...
+    - Only applicable to [classification models](../../models/pooling_models/classify.md).
+- Embedding Usages
+    - [Cohere Embed API](../../models/pooling_models/embed.md#cohere-embed-api) (`/v2/embed`)
+    - [OpenAI-compatible Embeddings API](../../models/pooling_models/embed.md#openai-compatible-embeddings-api) (`/v1/embeddings`)
+    - Only applicable to [embedding models](../../models/pooling_models/embed.md).
+- Scoring Usages
+    - [Score API](../../models/pooling_models/scoring.md#score-api) (`/score`)
+    - [Cohere Rerank API](../../models/pooling_models/scoring.md#rerank-api) (`/rerank`, `/v1/rerank`, `/v2/rerank`)
+    - Applicable to [score models](../../models/pooling_models/scoring.md) (cross-encoder, bi-encoder, late-interaction).
+- [pooling API](../../models/pooling_models/README.md#pooling-api) (`/pooling`)
```
Comment (DarkLight1337) [inline docs/serving/online_serving/README.md]:
```suggestion
- [Pooling API](../../models/pooling_models/README.md#pooling-api) (`/pooling`)
```

File: docs/serving/online_serving/README.md [hunk 19]
```diff
@@ -0,0 +1,182 @@
...
+
+## SageMaker APIs
+
+- `/invocations` - SageMaker-compatible endpoint (routes to the same inference functions as `/v1` endpoints)
+
+## Pooling APIs
+
+For further details on pooling models, please refer to [this page](../../models/pooling_models/README.md).
+
+- Classification Usages
```
Comment (DarkLight1337) [inline docs/serving/online_serving/README.md]: Maybe link to the corresponding model pages?

**Review by DarkLight1337 — APPROVED** [review]
LGTM now, thanks
